native/index.d.ts

```diff
@@ -0,0 +1,68 @@
+// <reference path="../react/react-native.d.ts" />
+
+declare module "styled-components/native" {
+    import * as ReactNative from "react-native";
+    import { StatelessComponent } from "react";
+
+    interface StyledFunction<T> {
+      (values: TemplateStringsArray): T;
+      (...values: any[]): T;
```

Comment:
On that note, maybe `StyledFunction` should have a type parameter for the props of each function?